proposals/flexible-vectors/BinaryFlexibleVectors.md

```diff
@@ -0,0 +1,313 @@
+
+Vector length:
+
+| Instruction              | `prefix` | `vecop` | Immediate operands       |
+| -------------------------|---------:|--------:|--------------------------|
+| `vec.i8.length`          |    `0x7A`|   `0x00`| -                        |
+| `vec.i16.length`         |    `0x79`|   `0x00`| -                        |
+| `vec.i32.length`         |    `0x78`|   `0x00`| -                        |
+| `vec.i64.length`         |    `0x77`|   `0x00`| -                        |
+| `vec.f32.length`         |    `0x76`|   `0x00`| -                        |
+| `vec.f64.length`         |    `0x75`|   `0x00`| -                        |
+
+Constructing vectors and accessing lanes:
+
+| Instruction              | `prefix` | `vecop` | Immediate operands       |
+| -------------------------|---------:|--------:|--------------------------|
+| `vec.i8.splat`           |    `0x7A`|   `0x10`| -                        |
+| `vec.i16.splat`          |    `0x79`|   `0x10`| -                        |
+| `vec.i32.splat`          |    `0x78`|   `0x10`| -                        |
+| `vec.i64.splat`          |    `0x77`|   `0x10`| -                        |
+| `vec.f32.splat`          |    `0x76`|   `0x10`| -                        |
+| `vec.f64.splat`          |    `0x75`|   `0x10`| -                        |
+| `vec.i8.extract_lane_u`  |    `0x7A`|   `0x11`| i:ImmLaneIdxV8           |
+| `vec.i16.extract_lane_u` |    `0x79`|   `0x11`| i:ImmLaneIdxV16          |
+| `vec.i32.extract_lane_u` |    `0x78`|   `0x11`| i:ImmLaneIdxV32          |
+| `vec.i64.extract_lane_u` |    `0x77`|   `0x11`| i:ImmLaneIdxV64          |
+| `vec.f32.extract_lane_u` |    `0x76`|   `0x11`| i:ImmLaneIdxV32          |
+| `vec.f64.extract_lane_u` |    `0x75`|   `0x11`| i:ImmLaneIdxV64          |
+| `vec.i8.extract_lane_s`  |    `0x7A`|   `0x12`| i:ImmLaneIdxV8           |
+| `vec.i16.extract_lane_s` |    `0x79`|   `0x12`| i:ImmLaneIdxV16          |
+| `vec.i32.extract_lane_s` |    `0x78`|   `0x12`| i:ImmLaneIdxV32          |
+| `vec.i64.extract_lane_s` |    `0x77`|   `0x12`| i:ImmLaneIdxV64          |
+| `vec.f32.extract_lane_s` |    `0x76`|   `0x12`| i:ImmLaneIdxV32          |
+| `vec.f64.extract_lane_s` |    `0x75`|   `0x12`| i:ImmLaneIdxV64          |
+| `vec.i8.replace_lane`    |    `0x7A`|   `0x13`| i:ImmLaneIdxV8           |
+| `vec.i16.replace_lane`   |    `0x79`|   `0x13`| i:ImmLaneIdxV16          |
+| `vec.i32.replace_lane`   |    `0x78`|   `0x13`| i:ImmLaneIdxV32          |
+| `vec.i64.replace_lane`   |    `0x77`|   `0x13`| i:ImmLaneIdxV64          |
+| `vec.f32.replace_lane`   |    `0x76`|   `0x13`| i:ImmLaneIdxV32          |
+| `vec.f64.replace_lane`   |    `0x75`|   `0x13`| i:ImmLaneIdxV64          |
+
+Shuffles:
+
+| Instruction              | `prefix` | `vecop` | Immediate operands       |
+| -------------------------|---------:|--------:|--------------------------|
+| `vec.i8.lshl`            |    `0x7A`|   `0x20`| -                        |
```

Comment:
what's the difference between lshl and shl? I think this is partially duplicated with the below section (Bitwise operations)

Reply:
`lshl` shifts lanes, and `shl` shifts bits within lanes.

Reply:
Ah okay, thanks!